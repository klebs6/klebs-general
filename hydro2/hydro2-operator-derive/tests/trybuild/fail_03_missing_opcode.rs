// ---------------- [ File: tests/trybuild/fail_03_missing_opcode.rs ]
//! Missing `opcode` in the `#[operator(...)]` attribute → compile_fail.
use hydro2_operator_derive::*;
use hydro2_operator::*; 
use named_item_derive::*;
use named_item::*;

#[derive(Debug,NamedItem,Operator)]
#[operator(
    execute="some_fn"
    // opcode is missing → should fail
)]
pub struct MissingOpcodeOp {
    name: String,
}

impl MissingOpcodeOp {

    pub async fn some_fn(&self) -> NetResult<()> {
        Ok(())
    }
}

fn main() {}
