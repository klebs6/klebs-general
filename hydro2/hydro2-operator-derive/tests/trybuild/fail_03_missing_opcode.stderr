error: No `opcode` specified in `#[operator(...)]`.
  --> tests/trybuild/fail_03_missing_opcode.rs:9:1
   |
9  | / #[operator(
10 | |     execute="some_fn"
11 | |     // opcode is missing â†’ should fail
12 | | )]
   | |__^
